definition_files.rst

```diff
@@ -223,6 +226,39 @@ Although the order of the sections in the def file is unimportant, they
 have been documented below in the order of their execution during the
 build process for ease of understanding.
 
+%arguments
+==========
+During the build process, variables defined via a matching pair of double curly
+brackets in the current definition file, i.e., the form of ``{{ variable }}``, 
+will be replaced by a value defined by a ``variable=value`` entry in this section.
+Note that values defined in this section are only the default values for defined
+variables. Any same variables with different values passed via build options
+``--build-arg`` or ``--build-arg-file`` will overwrite values defined in 
+this section. 
+
+Consider the example from the definition file above:
+
+.. code:: {command}
+
+   Bootstrap: docker
+   From: ubuntu:{{ VERSION }}
+   Stage: build
+
+   %arguments
+      VERSION=22.04
+
+The value in this entry ``VERSION=22.04`` will replace the template variable 
+``{{ VERSION }}`` during the build process. 
+
+If users build the image with the following commands:
+
+.. code:: {command}
+
+   apptainer build --build-arg VERSION=23.04 my_container.sif current.definition
+
+The base image tag will be set to version ``23.04`` rather than ``22.04``, which is 
+defined in the ``%arguments`` section. This rule also applies to build option ``--build-arg-file``.
```

Comment:
Change these two lines to just
```
then the From image tag will be overridden to ``23.04`` rather than ``22.04``.
```

I think the rest after that is redundant.

Reply:
thanks! I have reflected in this commit 
https://github.com/apptainer/apptainer-userdocs/pull/199/commits/cd74107eb7448c487d842141293b73f38fa0e2ce